### PR TITLE
InsertInto: support multiple source tables and destination tables

### PR DESCRIPTION
**Pull request by weiqingy**
Changes:
Added supports for multiple source tables and destination tables ( `InsertInto` queries). 

Tests:
New unit tests were added.
Manually tested in HDP cluster.

Case1: multiple source tables
```
sql(s"CREATE TABLE inputTable1 (a int, b string)")
sql(s"INSERT INTO inputTable1 VALUES(1, 'str1')")
sql(s"CREATE TABLE inputTable2 (c int, d string)")
sql(s"INSERT INTO inputTable2 VALUES(1, 'str2')")
sql(s"CREATE TABLE inputTable3 (e int, f string)")
sql(s"INSERT INTO inputTable3 VALUES(1, 'str3')")
sql(s"CREATE TABLE outputTable1 (a int, b string, c int, d string, e int, f string)")
sql("INSERT INTO outputTable1 " +
      "SELECT * FROM inputTable1, inputTable2, inputTable3 " +
      "where inputTable1.a = inputTable2.c AND inputTable1.a = inputTable3.e")
```

Case2: multiple destination tables
```
sql(s"create table inputTable4 (id int, code string, salary int)")
    sql(s"INSERT INTO inputTable4 VALUES(1, 'hihi', 100)")
    sql(s"create table outputTable2 (id int)")
    sql(s"create table outputTable3 (code string)")
    sql(s"create table outputTable4 (salary int)")
    sql(s"FROM inputTable4 " +
      s"INSERT INTO TABLE outputTable2 SELECT inputTable4.id " +
      s"INSERT INTO TABLE outputTable3 SELECT inputTable4.code " +
      s"INSERT into TABLE outputTable4 SELECT inputTable4.salary")
```

### CODE REVIEW

**Comment by weiqingy** [comment]
Hi @dongjoon-hyun , Could you please review this PR? Thanks.

File: spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala [hunk 1]
```diff
@@ -57,6 +67,22 @@ class InsertIntoHarvesterSuite extends FunSuite with Matchers with BeforeAndAfte
...
+    sparkSession.sql(s"CREATE TABLE $inputTable1 (a int, b string)")
+    sparkSession.sql(s"INSERT INTO $inputTable1 VALUES(1, 'str1')")
+    sparkSession.sql(s"CREATE TABLE $inputTable2 (c int, d string)")
+    sparkSession.sql(s"INSERT INTO $inputTable2 VALUES(1, 'str2')")
+    sparkSession.sql(s"CREATE TABLE $inputTable3 (e int, f string)")
+    sparkSession.sql(s"INSERT INTO $inputTable3 VALUES(1, 'str3')")
+    sparkSession.sql(s"CREATE TABLE $outputTable1 (a int, b string, c int, d string, e int, f string)")
+
+    // multiple destination tables
+    sparkSession.sql(s"create table $inputTable4 (id int, code string, salary int)")
```
Comment (dongjoon-hyun) [inline spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala]: Could you use an upper case for all SQL keyword?

File: spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala [hunk 2]
```diff
@@ -202,4 +228,77 @@ class InsertIntoHarvesterSuite extends FunSuite with Matchers with BeforeAndAfte
...
+      assert(pEntity.getAttribute("outputs").isInstanceOf[util.Collection[_]])
+      val outputs = pEntity.getAttribute("outputs").asInstanceOf[util.Collection[AtlasEntity]]
+      outputs.size() should be (1)
+      val outputTbl = outputs.asScala.head
+      outputTbl.getTypeName should be (external.HIVE_TABLE_TYPE_STRING)
+      outputTbl.getAttribute("name").toString should startWith ("output")
+      outputTbl.getAttribute(AtlasClient.REFERENCEABLE_ATTRIBUTE_NAME).toString should startWith (
+        "default.output")
+    })
+  }
```
Comment (dongjoon-hyun) [inline spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala]: Could you add another test case for `INSERT INTO MULTIPLE TABLES FROM MULTILE TABLE`?

File: spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkExecutionPlanTracker.scala [hunk 3]
```diff
@@ -66,7 +66,6 @@ class SparkExecutionPlanTracker(
     logWarn(s"Fail to execute query: {$qe}, {$funcName}", exception)
   }
 
-  // TODO: We should consider multiple inputs and multiple outs.
```
Comment (dongjoon-hyun) [inline spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkExecutionPlanTracker.scala]: It seems that we need to add test cases including both `view` and `table` to remove this TODO.

File: spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala [hunk 4]
```diff
@@ -57,6 +67,22 @@ class InsertIntoHarvesterSuite extends FunSuite with Matchers with BeforeAndAfte
```
Comment (dongjoon-hyun) [inline spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala]: Ur, can we use `DATABASE`?

At line 61, you can add the following.
```
sparkSession.sql("DROP DATABASE IF EXISTS SAC")
sparkSession.sql("CREATE DATABASE SAC")
sparkSession.sql("USE SAC")
```

And, add the following at `afterAll`.
```
sparkSession.sql("DROP DATABASE IF EXISTS SAC")
```

Comment (weiqingy) [reply spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala]: Done

File: spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala (hunk 2 above)
Comment (weiqingy) [reply spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala]: Done

File: spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala (hunk 1 above)
Comment (weiqingy) [reply spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala]: I have created a new issue to track this: https://github.com/hortonworks-spark/spark-atlas-connector/issues/17

Comment (weiqingy) [reply spark-atlas-connector/src/test/scala/com/hortonworks/spark/atlas/sql/InsertIntoHarvesterSuite.scala]: I've updated this. Also, I have created a new issue to track this: #17.

File: spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkExecutionPlanTracker.scala (hunk 3 above)
Comment (weiqingy) [reply spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/sql/SparkExecutionPlanTracker.scala]: I have created a new issue to track this: #17.

**Comment by dongjoon-hyun** [comment]
Ur, Travis CI seems to complain about test case failures.

**Comment by weiqingy** [comment]
It's strange. The latest PR did not touch anything related to `CREATE TABLE dest AS SELECT [] FROM source *`. Spark 2.3.1-SNAPSHOT may make some changes in SQL part.

**Comment by weiqingy** [comment]
I just ran the unit tests againt the master branch of SAC, and the unit tests in CreateHiveTableAsSelectHarvesterSuite failed.  
cc: @jerryshao

**Comment by weiqingy** [comment]
I've submitted a PR to fix the master branch issue: https://github.com/hortonworks-spark/spark-atlas-connector/pull/18

**Comment by dongjoon-hyun** [comment]
I merged #18. Could you rebase this?

**Comment by dongjoon-hyun** [comment]
I closed/reopened this to retrigger Travis CI.

**Pull request merged**